### DISCUSSION TITLE
Is there a limit number for Layer 4 Connection (TCP, QUIC)?

### DISCUSSION THREAD

**Discussion by elecbug** [libp2p/go-libp2p#2957] [Q&A] [answered]
Hello, we're a lab working on improving routing efficiency using libp2p.

We started the study with the aim of improving the L4(in particular, TCP) topology of the current libp2p by pre-investigating it, and when we collected logs and measured the topology, we saw that the TCP connection was tightly connected between almost all nodes.

There are three main questions.

1. Isn't the current go-libp2p implementation specifically limited to the maximum number of TCP connections? If so, is there a reason why it was made like that?
2. If you use the gossipsub protocol with TCP connection, does the gossipsub's metadata connection is the same role as a TCP connection?
3. I understand that go-libp2p has been applied to several projects. So, do other projects like ethereum maintain TCP connections that are proportional to the number of nodes participated? (If you don't think this question fits the forum, you don't have to answer it.)

Thanks.

**Comment by vyzo** [comment]
There is a ResourceManager that limits those things.

**Reply by elecbug** [reply]
If so, I understand number 3, and may I ask you to answer question number 2?

**Reply by vyzo** [reply, marked as answer]
It's a regular peer connection, nothing special about it.

**Reply by elecbug** [reply]
Thank you for your quick response
